### ISSUE TITLE
SidePanel component overlaps with header action button tooltips

### ISSUE THREAD

**Issue by modean** [carbon-design-system/ibm-products#4789]
## What package(s) are you using?

```
"@carbon/colors": "11.21.0",
"@carbon/ibm-products": "2.34.0",
"@carbon/layout": "11.21.0",
"@carbon/react": "1.54.0",
"@carbon/styles": "1.54.0",
"@carbon/themes": "11.34.0",
"@carbon/type": "11.26.0",
"react": "18.2.0",
"react-dom": "18.2.0",
"react-router-dom": "6.22.3"
```

## Detailed description

> Describe in detail the issue you're having.

If using the native `HeaderPanel` as provided by the Carbon UI shell components the tooltip of the header action button is visibal in both situations, if the `HeaderPanel` is collapsed and if the`HeaderPanel` is expanded.

For the `SidePanel` the behaviour is different. There the tooltip of the header action button is only visible if the `SidePaanel` is collapsed. If it is expanded the tooltip seems to be under the `SidePanel` component.

> Is this issue related to a specific component?

Yes the `SidePanel` component.

> What did you expect to happen? What happened instead? What would you like to
> see changed?

I would expect the behaviour of the `HeaderPanel` and `SidePanel` to be aligned wrt/ to those aspects that are similar or equivalent. I'd also argue that invisible tooltips my pose an accessibility issue even if a screenreader might still be able to read the invisible tooltip cause there are also persons w/ visual impairments that rely on the rendered visual contents.

> What browser are you working in?

Chrome 123.0.6312.105 (Official Build) (arm64)
Firefox 124.0.2 (64-bit)

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?

"@carbon/ibm-products": "2.34.0"

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

n/a

## Steps to reproduce the issue

I created a CodeSandbox [at the following location](https://codesandbox.io/p/devbox/createfullpagesample-9plsx3?file=%2Fsrc%2FApp.tsx%3A75%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cluocoy1i00073b6gweqod0r2%2522%252C%2522sizes%2522%253A%255B60%252C40%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cluocoy1h00023b6gm1qk9c77%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cluocoy1i00043b6gpfqz4y2j%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cluocoy1i00063b6gwnnik835%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B42.82810142949501%252C57.17189857050499%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cluocoy1h00023b6gm1qk9c77%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cluocoy1h00013b6gl2ktsvu7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cluocvpo600023b6gaex7i91z%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cluod7e8b00023b6g5ljvbv5e%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A75%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A75%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cluocoy1h00023b6gm1qk9c77%2522%252C%2522activeTabId%2522%253A%2522cluod7e8b00023b6g5ljvbv5e%2522%257D%252C%2522cluocoy1i00063b6gwnnik835%2522%253A%257B%2522id%2522%253A%2522cluocoy1i00063b6gwnnik835%2522%252C%2522activeTabId%2522%253A%2522cluod8cxy00693b6gkxxrjb5p%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522cluod8cxy00693b6gkxxrjb5p%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2523%2522%257D%255D%257D%252C%2522cluocoy1i00043b6gpfqz4y2j%2522%253A%257B%2522id%2522%253A%2522cluocoy1i00043b6gpfqz4y2j%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cluocoy1i00033b6ghnzu21m4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cluocoy1i00033b6ghnzu21m4%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) that replicates/reproduces the issue.

### HeaderPanel interaction

https://github.com/carbon-design-system/ibm-products/assets/659775/7238aabc-5b54-4c3d-ad6b-423f8637216c

### SidePanel interaction

https://github.com/carbon-design-system/ibm-products/assets/659775/bea108d2-28f2-4f69-86b8-63fed21acd38

**Comment by makafsal** [comment]
@matthewgallo @elycheea - The z-index is taken from the Carbon styles utilities file:

![image](https://github.com/carbon-design-system/ibm-products/assets/9197089/45f7e5be-bfab-4ae8-9bf4-73d284751a5e)


![image](https://github.com/carbon-design-system/ibm-products/assets/9197089/de3ddf2e-5cd8-48df-b465-de8b2ae0f1e6)

Here tooltip uses the value `floating: 6000` and the modal uses the value `modal: 8000`, this is causing this issue.  Is there any way to fix this?

**Comment by modean** [comment]
As another remark I want to point out the following. If for instance implementing an about menu or a profile menu using an action button next tot he help action button on the Carbon header, one needs to know that `c4p--page-header` has an arbitrary `z-index` of `7999`.

The about menu or user profile menu thus needs to have a `z-index` greater than `7999` cause otherwise `c4p--page-header` would float atop of that said about menu or user profile menu.

In other words I found it a bit tricky to handle all the various `z-index` values to get the custom menu I implemented right also cause the docs IIRC do not call out that there's a correlation between the various `z-index` settings.
